### ISSUE TITLE
proposition for alternative property for InProgressDataDate column in MARCO-BOLO_Metadata_Dataset_Record_WP5

### ISSUE THREAD

**Issue by laurianvm** [marco-bolo/metadata-catalogue#28]
[MARCO-BOLO_Metadata_Dataset_Record_WP5](https://docs.google.com/spreadsheets/d/1jH8Gp50y9w_SsoFELYTKV6ohPlkO3nqm/edit?gid=115560455#gid=115560455)

Duplicate keys are not allowed in json(-ld)
--> meaning that the schema.org equivalent property of 'additionalProperty' for the columns "InProgressDataDate" and "EVDescription" needs to change for one these columns;
--> propose to change 'additionalProperty' into 'datePublished' for  "InProgressDataDate"  

Is this okay?

**Comment by kmexter** [comment]
fine by me - would need to change for all the WP spreadsheets, mind

**Comment by kmexter** [comment]
except ... these googlesheets do not need to be transformed, laurian. only the URIs they refer to.

**Comment by laurianvm** [comment]
only the value in cell D8 of the 'Column descriptions' tab would have to be changed (no other values need changing) - and this would only be to be aligned with what is actually used in the triples (but not required for things to work/not break)

**Comment by kmexter** [comment]
Are we OK here? In that, the "in progress date" is not something you need to read into your templates. Can I close this?

**Comment by laurianvm** [comment]
with https://github.com/marco-bolo/dataset-catalogue/issues/36 duplicate keys issue disappeared on its own

however, https://schema.org/additionalProperty is not expected to be used on Dataset, 
hence proposition to use:
https://github.com/marco-bolo/dataset-catalogue/blob/9e95ff3052e4a4b34d3116dbcfacca6ee5f2960f/scripts/tests/templates/gsheet.jsonld.ldt.j2#L404-L406

@pieterprovoost or @pbuttigieg could you indicate whether schema.org-model needs to be rigidly followed? and which predicate is preferred in this case?

**Comment by kmexter** [comment]
yeh, this is for @pieterprovoost or @pbuttigieg - not my expertise. I suggest that you two P's go thru the list of properties - col D of https://docs.google.com/spreadsheets/d/1jH8Gp50y9w_SsoFELYTKV6ohPlkO3nqm/edit?gid=115560455#gid=115560455 for example - and double check them

**Comment by pbuttigieg** [comment]
> however, https://schema.org/additionalProperty is not expected to be used on Dataset, 

It does validate and ODIS does use it on other types (we so reason for limiting its range). 

That being said, there may be a better way. 


> hence proposition to use:
> https://github.com/marco-bolo/dataset-catalogue/blob/9e95ff3052e4a4b34d3116dbcfacca6ee5f2960f/scripts/tests/templates/gsheet.jsonld.ldt.j2#L404-L406
> 
> @kmexter , @pieterprovoost or @pbuttigieg could you indicate whether schema.org-model needs to be rigidly followed? and which predicate is preferred in this case?
> 


"For data that are still in progress; please fill in an approximate date (YYYY-MM) you expect them to be published"

This says to be that schema:`[creativeWorkStatus](https://schema.org/creativeWorkStatus)` should be set to something like "Incomplete".

If the value of creativeWorkStatus uses a DefinedTerm type, you can add the expected date of release in a description property of DefinedTerm.

**Comment by pbuttigieg** [comment]
@laurianvm something like the following is good form for all creative works (datasets)


```json
"creativeWorkStatus": {
           "@type": "DefinedTerm",
           "name": "Incomplete",         
           "identifier": "http://purl.obolibrary.org/obo/NCIT_C49160",
            "inDefinedTermSet": "http://purl.obolibrary.org/obo/ncit.owl",
            "termCode": "NCIT:C49160"
           
        }
```

**Comment by kmexter** [comment]
these would then only be for the jsonld files filled with info in the googlesheets, clearly not for something from a DOI. The maintainer would be the agent mentioned in the DatasetDescriber column
Tho...no-one has filled this column in, so you wont actually get any values in these stanzas

**Comment by kmexter** [comment]
all very true
one comment makes me think: we have not created a template for how to describe an aggregate dataset - which indeed, we *will* be creating in MBO. Does this merit a new issue?

**Comment by pbuttigieg** [comment]
> all very true
> one comment makes me think: we have not created a template for how to describe an aggregate dataset - which indeed, we *will* be creating in MBO. Does this merit a new issue?

Yes, I'll create one. A Dataset is of arbitrary size and mereology, so that type can be used unless a DataCatalog is in play.

**Comment by laurianvm** [comment]
being followed up in #46, so closing this one
